File: core/domain/src/test/kotlin/com/deepseek/harness/android/domain/model/SessionSummaryTest.kt
package com.deepseek.harness.android.domain.model

import org.junit.Assert.assertEquals
import org.junit.Test

class SessionSummaryTest {

    @Test
    fun `display title uses durable title first`() {
        val session = SessionSummary(
            id = "session-1",
            title = "Triage issue",
            cwd = "/home/user/work",
        )

        assertEquals("Triage issue", session.displayTitle)
    }

    @Test
    fun `display title falls back to workspace basename then id`() {
        val byPath = SessionSummary(
            id = "session-1",
            cwd = "/home/user/checkout",
        )
        val withTrailingSeparator = SessionSummary(
            id = "session-2",
            cwd = "C:\\work\\checkout\\",
        )
        val withoutAncestors = SessionSummary(
            id = "session-3",
        )

        assertEquals("checkout", byPath.displayTitle)
        assertEquals("checkout", withTrailingSeparator.displayTitle)
        assertEquals("session-3", withoutAncestors.displayTitle)
    }
}

File: flutter/packages/domain/test/model/session_summary_test.dart
import 'package:test/test.dart';

import 'package:domain/model/session.dart';

void main() {
  test('display title uses durable title first', () {
    const session = SessionSummary(
      id: 'session-1',
      title: 'Triage issue',
      cwd: '/home/user/work',
    );

    expect(session.displayTitle, 'Triage issue');
  });

  test('display title falls back to workspace basename then id', () {
    const byPath = SessionSummary(
      id: 'session-1',
      cwd: '/home/user/checkout',
    );
    const withTrailingSeparator = SessionSummary(
      id: 'session-2',
      cwd: 'C:\\work\\checkout\\',
    );
    const withoutAncestors = SessionSummary(
      id: 'session-3',
    );

    expect(byPath.displayTitle, 'checkout');
    expect(withTrailingSeparator.displayTitle, 'checkout');
    expect(withoutAncestors.displayTitle, 'session-3');
  });
}
